### PR TITLE
refactor(radio): implement design tokens spec

### PR DESCRIPTION
**Pull request by lalithkarikelli**
## Description

`<hx-radio>` : Added design tokens

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF2204

## Before you request a review for this PR:

- [x] For UI changes, did you manually test in recent versions of modern browsers (Chrome, Firefox, and Safari)?
- [x] For UI changes, did you manually test in IE11 and legacy Edge?
- [ ] Did you add component tests for any new code?
- [ ] Did you run the component unit tests via `yarn test` to ensure all tests passed?
- [ ] Did you include a screenshot of the component tests?
- [ ] If you changed/added functionality, did you update the demo page and documentation?
- [ ] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] In Jira, have you linked to this PR on the ticket(s)?

### CODE REVIEW

**Review by 100stacks — CHANGES_REQUESTED** [review]
Let's hold off on removing states for now.

File: src/scss/components/radio/_index.scss [hunk 1]
```diff
@@ -46,10 +46,10 @@ hx-radio-control {
   // facade + text
   > label {
     // Radio v2.0 specs
-    font-size: 1rem;
+    font-size: $rax-component-radio-label-default-font-size;
     color: $gray-950;
-    color: var(--hxRadio-radioControl-layout-label-color, $gray-950);
-    font-weight: 400; // normal
+    color: $rax-component-radio-label-default-color; // in tokens it is $gray-900
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/_index.scss:50]: Changes have been done, please review.

File: src/scss/components/radio/_index.scss [hunk 2]
```diff
@@ -101,10 +101,10 @@ hx-radio-control {
 hx-radio-control {
   > label > hx-radio {
     background-color: $gray-0;
-    background-color: var(--hxRadio-pristine-radio-bgcolor, $gray-0);
-    border: 1px solid currentColor;
+    background-color: $rax-component-radio-default-background-color;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/_index.scss:103]: Changes have been done, please review.

File: src/scss/components/radio/_index.scss [hunk 3]
```diff
@@ -101,10 +101,10 @@ hx-radio-control {
...
+    border: $rax-component-radio-base-border-width solid currentColor;
     color: $gray-500;
-    color: var(--hxRadio-pristine-radio-color, $gray-500);
+    color: $rax-component-radio-default-border-color; // in tokens it is border-color
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/_index.scss:106]: Changes have been done, please review.

File: src/scss/components/radio/_index.scss [hunk 4]
```diff
@@ -128,24 +128,24 @@ hx-radio-control {
 
     &:checked {
       color: $blue-700;
-      color: var(--hxRadio-pristine-radio-checked-color, $blue-700);
+      color: $rax-component-radio-checked-border-color;// in tokens it is border-color
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/_index.scss]: Changes have been done, please review.

File: src/scss/components/radio/_index.scss [hunk 5]
```diff
@@ -128,24 +128,24 @@ hx-radio-control {
...
 
       ~ label > hx-radio {
-        border: 1px solid currentColor;
+        border: $rax-component-radio-base-border-width solid currentColor;
         color: $blue-700;
-        color: var(--hxRadio-pristine-radio-checked-label-color, $blue-700);
+        color: $rax-component-radio-checked-border-color;// no specific token for checked label color
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/_index.scss]: Changes have been done, please review.

File: src/scss/components/radio/_index.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/_index.scss:50]: Changes have been done, please review.

File: src/scss/components/radio/_index.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/_index.scss:103]: Changes have been done, please review.

File: src/scss/components/radio/_index.scss (hunk 4 above)
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/_index.scss]: Changes have been done, please review.

File: src/scss/components/radio/_index.scss (hunk 5 above)
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/_index.scss]: Changes have been done, please review.

File: src/scss/components/radio/mixins/_hxRadioControl.scss [hunk 6]
```diff
@@ -6,22 +6,22 @@
     box-shadow: $focus-glow;
   } @else if $pseudo-state == hover {
     color: $blue-700;
-    color: var(--hxRadio-pristine-radio-hover-color, $blue-700);
+    color: $rax-component-radio-hover-border-color;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/mixins/_hxRadioControl.scss:8]: Changes have been done, please review.

File: src/scss/components/radio/mixins/_hxRadioControl.scss [hunk 7]
```diff
@@ -6,22 +6,22 @@
...
 
     &::-ms-check {
       background-color: $gray-0;
-      background-color: var(--hxRadio-pristine-radio-hover-ie-bgcolor, $gray-0);
+      background-color: $rax-component-radio-hover-background-color;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/mixins/_hxRadioControl.scss:12]: Changes have been done, please review.

File: src/scss/components/radio/mixins/_hxRadioControl.scss [hunk 8]
```diff
@@ -6,22 +6,22 @@
...
       color: $blue-700;
-      color: var(--hxRadio-pristine-radio-hover-ie-color, $blue-700);
+      color: $rax-component-radio-hover-border-color;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/mixins/_hxRadioControl.scss:14]: Changes have been done, please review.

File: src/scss/components/radio/mixins/_hxRadioControl.scss [hunk 9]
```diff
@@ -6,22 +6,22 @@
...
     }
   } @else {
     color: $gray-500;
-    color: var(--hxRadio-pristine-radio-color, $gray-500);
+    color: $rax-component-radio-default-border-color;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/mixins/_hxRadioControl.scss:18]: Changes have been done, please review.

File: src/scss/components/radio/mixins/_hxRadioControl.scss [hunk 10]
```diff
@@ -6,22 +6,22 @@
...
 
     &::-ms-check {
       background-color: $gray-0;
-      background-color: var(--hxRadio-pristine-radio-ie-bgcolor, $gray-0);
-      border: 1px solid currentColor;
+      background-color: $rax-component-radio-default-background-color;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/mixins/_hxRadioControl.scss:22]: Changes have been done, please review.

File: src/scss/components/radio/mixins/_hxRadioControl.scss [hunk 11]
```diff
@@ -31,20 +31,12 @@
...
-      background-color: $red-status-100;
-      background-color: var(--hxRadio-invalid-radio-hover-ie-bgcolor, $red-status-100);
-      border: 1px solid currentColor;
-      color: $red-status-500;
-      color: var(--hxRadio-invalid-radio-hover-ie-color, $red-status-500);
-    }
   } @else {
     color: $red-status-500;
-    color: var(--hxRadio-invalid-radio-color, $red-status-500);
+    color: $rax-component-radio-invalid-border-color;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/mixins/_hxRadioControl.scss:43]: Changes have been done, please review.

File: src/scss/components/radio/mixins/_hxRadioControl.scss [hunk 12]
```diff
@@ -53,25 +45,17 @@
...
-      background-color: $gray-100;
-      background-color: var(--hxRadio-disabled-radio-hover-ie-bgcolor, $gray-100);
-      border: 1px solid currentColor;
-      color: $gray-500;
-      color: var(--hxRadio-disabled-radio-hover-ie-color, $gray-500);
-    }
   } @else {
     color: $gray-500;
-    color: var(--hxRadio-disabled-radio-color, $gray-500);
+    color: $rax-component-radio-disabled-border-color;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/mixins/_hxRadioControl.scss:65]: Changes have been done, please review.

File: src/scss/components/radio/mixins/_hxRadioFacade.scss [hunk 13]
```diff
@@ -7,10 +7,10 @@
   }
   @else {
     background-color: $gray-0;
-    background-color: var(--hxRadio-pristine-radio-facade-bgcolor, $gray-0);
-    border: 1px solid currentColor;
+    background-color: $rax-component-radio-default-background-color;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/mixins/_hxRadioFacade.scss:9]: Changes have been done, please review.

File: src/scss/components/radio/mixins/_hxRadioFacade.scss [hunk 14]
```diff
@@ -19,35 +19,19 @@
...
-  }
-  @else {
-    border: 1px solid currentColor;
-    color: $gray-500;
-    color: var(--hxRadio-disabled-radio-facade-color, $gray-500);
-    cursor: not-allowed;
-  }
+  border: $rax-component-radio-base-border-width solid currentColor;
+  color: $gray-500;
+  color: $rax-component-radio-disabled-border-color;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/mixins/_hxRadioFacade.scss:40]: Changes have been done, please review.

File: docs/components/radio/test.html [hunk 15]
```diff
@@ -181,10 +181,6 @@ <h2>Optional - Disabled</h2>
             <td>{{radioSet('pristine', isFocused=true, isDisabled=true)}}</td>
             <td>N/A</td>
           </tr>
-          <tr>
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply docs/components/radio/test.html]: Changes have been done, please review.

File: src/scss/components/radio/_index.scss [hunk 16]
```diff
@@ -172,15 +172,6 @@ hx-radio-set.hxInvalid input[type="radio"]:enabled {
...
   }
-
-  &:hover {
-    @include hxRadioControl(invalid, hover);
-  }
-
-  // facade styles
-  ~ label > hx-radio:hover {
-    @include hxRadioFacade(invalid, hover);
-  }
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/_index.scss]: Changes have been done, please review.

File: src/scss/components/radio/_index.scss [hunk 17]
```diff
@@ -191,22 +182,13 @@ hx-radio-control > input[type="radio"]:disabled {
     @include hxRadioControl(disabled, focus);
   }
 
-  &:hover {
-    @include hxRadioControl(disabled, hover);
-  }
-
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/_index.scss]: Changes have been done, please review.

File: src/scss/components/radio/_index.scss [hunk 18]
```diff
@@ -191,22 +182,13 @@ hx-radio-control > input[type="radio"]:disabled {
...
-    color: var(--hxRadio-disable-radio-label-color, $gray-600);
+    color: $rax-component-radio-label-disabled-color;
 
     > hx-radio {
       @include hxRadioFacade(disabled);
-
-      &,
-      &:hover {
-        @include hxRadioFacade(disabled, hover);
-      }
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/_index.scss:209]: Changes have been done, please review.

File: src/scss/components/radio/mixins/_hxRadioControl.scss [hunk 19]
```diff
@@ -31,20 +31,12 @@
...
   @if $pseudo-state == focus {
     box-shadow: $focus-glow-invalid;
-  } @else if $pseudo-state == hover {
-    &::-ms-check {
-      background-color: $red-status-100;
-      background-color: var(--hxRadio-invalid-radio-hover-ie-bgcolor, $red-status-100);
-      border: 1px solid currentColor;
-      color: $red-status-500;
-      color: var(--hxRadio-invalid-radio-hover-ie-color, $red-status-500);
-    }
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/mixins/_hxRadioControl.scss]: Changes have been done, please review.

File: src/scss/components/radio/mixins/_hxRadioControl.scss [hunk 20]
```diff
@@ -53,25 +45,17 @@
...
   @if $pseudo-state == focus {
     box-shadow: $focus-glow-disabled;
-  } @else if $pseudo-state == hover {
-    &::-ms-check {
-      background-color: $gray-100;
-      background-color: var(--hxRadio-disabled-radio-hover-ie-bgcolor, $gray-100);
-      border: 1px solid currentColor;
-      color: $gray-500;
-      color: var(--hxRadio-disabled-radio-hover-ie-color, $gray-500);
-    }
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/mixins/_hxRadioControl.scss]: Changes have been done, please review.

File: src/scss/components/radio/mixins/_hxRadioFacade.scss [hunk 21]
```diff
@@ -19,35 +19,19 @@
   @if $pseudo-state == focus {
     box-shadow: $focus-glow-invalid;
   }
-  @else if $pseudo-state == hover {
-    background-color: $red-status-100;
-    background-color: var(--hxRadio-invalid-radio-facade-hover-bgcolor, $red-status-100);
-    border: 1px solid currentColor;
-    color: $red-status-500;
-    color: var(--hxRadio-invalid-radio-facade-hover-color, $red-status-500);
-  }
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/mixins/_hxRadioFacade.scss]: Changes have been done, please review.

File: src/scss/components/radio/mixins/_hxRadioFacade.scss [hunk 22]
```diff
@@ -19,35 +19,19 @@
...
-    border: 1px solid currentColor;
-    color: $gray-500;
-    color: var(--hxRadio-disabled-radio-facade-hover-color, $gray-500);
-  }
-  @else {
-    border: 1px solid currentColor;
-    color: $gray-500;
-    color: var(--hxRadio-disabled-radio-facade-color, $gray-500);
-    cursor: not-allowed;
-  }
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/radio/mixins/_hxRadioFacade.scss:50]: Changes have been done, please review.

**Comment by lalithkarikelli** [comment]
Rebased with master and squashed all commits

**Review by badejayayesubabu — APPROVED** [review]
DEV LGTM

**Review by 100stacks — APPROVED** [review]
Dev LGTM

**Pull request merged**